File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.js
function solution(n, words) {
    const answer = [];
    
    words.forEach((word, index) => {
        if (index === 0 || answer.length) return;
        
        const endWord = words[index - 1].at(-1);
        const startWord = word[0];
        
        if (endWord !== startWord) {
            const num = index % n + 1;
            const order = index % n === 0 ? Math.ceil(index + 1 / n) : Math.ceil(index / n);
            
            answer.push(num);
            answer.push(order);
            return;
        }
        
        const isDuplicated = words.slice(0, index).indexOf(word) !== -1;
        
        if (isDuplicated) {
            const num = index % n + 1;
            const order = Math.floor(index / n) + 1;
            
            answer.push(num);
            answer.push(order);
            return;
        }
    })
    
    if (!answer.length) {
        answer.push(0);
        answer.push(0);
    }
    
    return answer;
}

File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.py
def solution(n, words):
    answer = [0, 0]
    count = [0] * n
    word_list = []
    order = -1
    
    for i in range(len(words)):
        order += 1
        order %= n
        
        if i > 0 and (words[i] in word_list or words[i - 1][-1] != words[i][0]):
            answer = [order + 1, i // n + 1]
            break
            
        word_list.append(words[i])
        
    return answer